plugins/modules/docker_plugin.py

```diff
@@ -346,11 +346,18 @@ def disable(self):
 
     @property
     def result(self):
+        plugin_data = {}
+        if self.parameters.state != 'absent':
+            try:
+                plugin_data = self.client.get_json('/plugins/{0}/json', self.preferred_name)
+            except NotFound:
+                # This can happen in check mode
+                pass
         result = {
             'actions': self.actions,
             'changed': self.changed,
             'diff': self.diff,
-            'plugin': self.client.get_json('/plugins/{0}/json', self.preferred_name) if self.parameters.state != 'absent' else {}
+            'plugin': plugin_data
```

Comment:
optional nit/opinionated style preference
```suggestion
            'plugin': plugin_data,
```

Reply:
Damn, I completely forgot while being in a hurry... :) Thanks!